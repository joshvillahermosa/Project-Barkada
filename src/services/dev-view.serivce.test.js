import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import { Button } from 'react-native-elements';
import { createDevButton } from './dev-view.service';

describe('Dev View Service', () => {
  describe('Creating Dev View Buttons', () => {
    const devViewButtonConfig = [{
      name: 'View One',
      view: null,
      enabled: false,
    }, {
      name: 'View One',
      view: null,
      enabled: true,
    }]

    it('Should render button components based on configs', () => {
      const mapFunction = createDevButton(React, Button, '#000');
      const tree = renderer.create(<View>{devViewButtonConfig.map(mapFunction)}</View>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
