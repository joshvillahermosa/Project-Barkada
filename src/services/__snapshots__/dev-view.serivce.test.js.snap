// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dev View Service Creating Dev View Buttons Should render button components based on configs 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#000",
            },
            undefined,
            false,
            undefined,
            undefined,
            undefined,
            Object {
              "marginBottom": 10,
            },
            Object {
              "backgroundColor": "#dadee0",
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                false,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          View One
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#000",
            },
            undefined,
            false,
            undefined,
            undefined,
            undefined,
            Object {
              "marginBottom": 10,
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                false,
                undefined,
                undefined,
                undefined,
                undefined,
                false,
              ],
            ]
          }
        >
          View One
        </Text>
      </View>
    </View>
  </View>
</View>
`;
